docs/cudf/source/cudf_pandas/faq.md

```diff
@@ -142,6 +142,52 @@ cuDF (learn more in [this
 blog](https://medium.com/rapids-ai/easy-cpu-gpu-arrays-and-dataframes-run-your-dask-code-where-youd-like-e349d92351d)) and the [RAPIDS Accelerator for Apache Spark](https://nvidia.github.io/spark-rapids/)
 provides a similar configuration-based plugin for Spark.
 
+## How do I know if an object is a `cudf.pandas` proxy object?
+
+To determine if an object is a `cudf.pandas` proxy object, you can use the `is_cudf_pandas_obj` API. This function checks if the given object is a proxy object that wraps either a `cudf` or `pandas` object. Here is an example of how to use this API:
+
+```python
+from cudf.pandas import is_cudf_pandas_obj
+
+obj = ...  # Your object here
+if is_cudf_pandas_obj(obj):
+    print("The object is a cudf.pandas proxy object.")
+else:
+    print("The object is not a cudf.pandas proxy object.")
+```
+
+There are APIs to detect `Series`, `DataFrame`, `Index`, and `ndarray` objects separately:
+
+* `is_cudf_pandas_series`: Detects if the object is a `cudf.pandas` proxy `Series`.
+* `is_cudf_pandas_dataframe`: Detects if the object is a `cudf.pandas` proxy `DataFrame`.
+* `is_cudf_pandas_index`: Detects if the object is a `cudf.pandas` proxy `Index`.
+* `is_cudf_pandas_ndarray`: Detects if the object is a `cudf.pandas` proxy `ndarray`.
+
+## How can I access the underlying GPU or CPU objects?
+
+When working with `cudf.pandas` proxy objects, it is sometimes necessary to get true `cudf` or `pandas` objects that reside on GPU or CPU.
+For example, this can be used to ensure that GPU-aware libraries that support both `cudf` and `pandas` can use the `cudf`-optimized code paths that keep data on GPU when processing `cudf.pandas` objects.
+Otherwise, the library might use less-optimized CPU code because it thinks that the `cudf.pandas` object is a plain `pandas` dataframe.
+
+The following methods can be used to retrieve the actual `cudf` or `pandas` objects:
+
+- `as_gpu_object()`: This method returns the `cudf` object from the proxy.
+- `as_cpu_object()`: This method returns the `pandas` object from the proxy.
+
+If `as_gpu_object()` is called on a proxy array, it will return a `cupy` array and `as_cpu_object` will return a `numpy` array.
```

Comment:
What happens if it is called on something that is not a proxy type? If it is always the user's responsibility to check with the functions above, we should document that. Alternatively, we could do something like return `None` if it is not a proxy to support usage like:
```
if (ser := as_gpu_object(obj)) is not None:
    # Do unwrapped processing
```

but that feels unnecessarily complex.

Reply:
That's why we can use the `isinstance...` check we are introducing here no?

Reply:
Yes I'm just wondering if there is a way to make this a one-step process, but I agree it ends up basically the same and doesn't improve much while adding complexity. No change required here.